arrow-core-data/src/main/kotlin/arrow/core/MapK.kt

```diff
@@ -25,14 +25,14 @@ data class MapK<K, out A>(private val map: Map<K, A>) : MapKOf<K, A>, Map<K, A>
 
   fun <B, Z> ap2(f: MapK<K, (A, B) -> Z>, fb: MapK<K, B>): Map<K, Z> =
     f.map.flatMap { (k, f) ->
-      this.flatMap { a -> fb.flatMap { b -> mapOf(Tuple2(k, f(a, b))).k() } }
-        .getOption(k).map { Tuple2(k, it) }.k().asIterable()
-    }.k()
+      this.flatMap { a -> fb.flatMap { b -> mapOf(Tuple2(k, f(a, b))).k() } }[k]
+        ?.let { Pair(k, it) }.asIterable()
```

Comment:
```suggestion
        ?.let { Tuple2(k, it) }.asIterable()
```

Reply:
@nomisRev suggested we switch to pair for this one. The impression I got is that Pair is slightly more efficient than Tuple2 in this case.

Reply:
Using `Pair` allows using Kotlin's standard library functions which we'd otherwise still have to provide. I thought we'd prefer Kotlin Std.